app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/ResponseCustomizer.java

```diff
@@ -57,6 +57,37 @@ public void setOutputDataShape(final DataShape dataShape) {
         isUnifiedDataShape = isUnifiedDataShape(dataShape);
     }
 
```

Comment:
At what point can we remove checking the legacy way? Should we add a deprecate warning?

Reply:
The data shapes could be baked into Integration (like the model), I'd prefer to drop it, but I'm unsure if the integrations stop working at that point. It depends on how many integrations are living with the data shapes in JSON schema without the `id`.

Reply:
You're saying to support legacy forever?

Reply:
@KurtStam what's your suggestion?

![](http://thegoagencyusa.com/wp-content/uploads/2015/06/tag-would-you-rather-5.jpg)

Reply:
Actually, I just realized we added this in 1.5, so removing it won't impact anyone really. I'm going to drop it.